databases/tiledb/files/patch-linking.diff

```diff
@@ -0,0 +1,47 @@
+--- tiledb/CMakeLists.txt.orig	2023-07-08 03:01:33.000000000 +0800
++++ tiledb/CMakeLists.txt	2023-07-10 17:52:17.000000000 +0800
+@@ -638,6 +638,7 @@
+     ZLIB::ZLIB
+     Zstd::Zstd
+     libmagic
++    MacportsLegacySupport
+ )
+ if (TILEDB_VCPKG)
+   # vcpkg uses SPDLOG_FMT_EXTERNAL=1
+@@ -654,6 +655,13 @@
+   )
+ endif()
+ 
++if(CMAKE_OSX_ARCHITECTURES STREQUAL "ppc")
++  target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
++    INTERFACE
++    atomic
```

Comment:
the need for the atomic library here has to do with gcc being used as the compiler, not related to ppc as the arch.

Reply:
I have been told a number of times that `i386` does not need it. Is that wrong?

64-bit archs do not need it with GCC. (Yes, I know that we usually do not bother to write a complete condition, since few if anyone gonna use GCC on x86_64 or aarch64, and no one uses ppc64, or if they do, they can survive with a redundant linking.)

Reply:
well, last I tried it. things can change. I'd have to run a test with gcc building i386 using atomics to stake my life on it.

Reply:
Ok, I make no claims re `i386` here. What happened is that I mentioned it a number of times in different repos that 32-bit archs need linking to `libatomic` (with GCC), but got replies that `i386` does not. It could have been that persons commenting only used Clang and assumed it extrapolates to GCC, or w/e.

Reply:
nothing has changed. All gcc builds fail, 32bit and 64 bit, on Intel archs unless libatomic is added.

So all those folks were off base.

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81358

Reply:
@kencu If I understand correctly, @iains says only 32-bit needs it:

> There are no 64 bit (or larger) lock-free atomics on PPC 32 - this is a constraint of the ISA, not the software.

See: https://github.com/iains/darwin-toolchains-start-here/discussions/30#discussioncomment-4242172

Reply:
would never disagree with Iain, but this code fails on 64bit Intel unless libatomic is added, so --- 

```
#include <stdio.h>
#include <complex.h>
#include <stdatomic.h>

int main(void)
{
    _Atomic long double _Complex x = 0.0 + 0.0*I;
    for (int i=0; i<1000; i++) {
        x += 1.0;
    }
    printf("(%Lf,%Lf)\n",creall(x), cimagl(x));
    printf("%zu\n",sizeof(x));
    return 0;
}

$ gcc-7 -std=c11 a.c -latomic
$ gcc-7 -std=c11 a.c         
Undefined symbols for architecture x86_64:
  "___atomic_feraiseexcept", referenced from:
      _main in cceeEpua.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
```

(I also tried newer gccs).

Reply:
> this basic info is that the need for libatomic is imposed by using gcc

@kencu Then I do not really get why upstream does not fix this. If every arch requires `libatomic` and not just non-Intel 32-bit ones, the library must be linked unconditionally.

Sure, then let us change it to be used with any GCC build. We certainly do not want to have x86_64 or ppc64 broken. (I may not be interested to invest significant efforts in fixing Intel builds specifically, but I do prefer them fixed rather than not.)

Reply:
(I'm not sure, exactly, how the presence or absence of 64 bit lock free atomics on PPC32 is relevant to needing libatomic when building on Intel anyway, but ... )